api/cases/case-list.js

```diff
@@ -13,67 +23,115 @@ function getOnlineHearing(caseIds, options) {
-                            if(state!= undefined && state!= null && state.state_name!= undefined && state.state_name != null) {
+                            if (state != undefined && state != null && state.state_name != undefined && state.state_name != null) {
                                 caseRow.status = format(state.state_name);
-                                if(new Date(caseRow.last_modified) < new Date(state.state_datetime)) {
+                                if (new Date(caseRow.last_modified) < new Date(state.state_datetime)) {
                                     caseRow.last_modified = state.state_datetime;
                                 }
                             }
                         });
                     }
-                    return caseData;
+                    resolve(casesData);
                 });
-                return casesData;
-            })
-        .then(casesData => {
-        const results = rawCasesReducer(casesData, sscsCaseListTemplate.columns)
-            .filter(row => row.case_reference !== undefined && row.case_reference !== null && row.case_reference.trim().length > 0)
-            .sort(function (result1, result2) {
-            return new Date(result1.case_fields.dateOfLastAction) - new Date(result2.case_fields.dateOfLastAction);
-        });
-        const aggregatedData = {...sscsCaseListTemplate, results: results};
-        res.setHeader('Access-Control-Allow-Origin', '*');
-        res.setHeader('content-type', 'application/json');
-        res.status(200).send(JSON.stringify(aggregatedData));
-    }).catch(response => {
+        }
+        else {
+            resolve(casesData);
+        }
+    });
+}
+
+
+
+
+function processCaseList(caseList, options) {
+    return new Promise((resolve, reject) => {
+        if (caseList && caseList.length) {
+            getCOR(caseList, options).then(casesData => {
+                const jurisdiction = casesData[0].jurisdiction;
+                const caseType = casesData[0].case_type_id;
+                const template = getListTemplate(jurisdiction, caseType);
+                const results = rawCasesReducer(casesData, template.columns)
+                    .filter(row => !!row.case_fields.case_ref)
+                    .sort(function (result1, result2) {
+                        return new Date(result1.case_fields.dateOfLastAction) - new Date(result2.case_fields.dateOfLastAction);
+                    });
+
+                resolve(results)
+            });
+        }
+        else {
+            resolve([]);
+        }
+    });
+}
+
+
+function combineLists(lists) {
+    return [].concat(...lists);
+}
+
+
+//List of cases
+module.exports = (req, res, next) => {
+    const token = req.auth.token;
+    const userId = req.auth.userId;
+    const options = {
+        headers: {
+            'Authorization': `Bearer ${token}`,
+            'ServiceAuthorization': req.headers.ServiceAuthorization
+        }
+    };
+
+    const jurisdictions = [{
```

Comment:
please add a comment above this. explaining that this is a temporary solution and add the jira ticket link with the long term solution - ie. getting jur and hascor from idam rather than hardcoding it.

Reply:
please do this.

Reply:
There is no ticket for this. This PR mentions quite clearly that this is not a finalised perfect solution

Reply:
@mike-suggitt we really should try to help the team - exchange knowledge. This is our last day. It would be super helpful for the team to have a clear understanding of what still needs doing in order to get multi jur working.

Reply:
just fyi @cksanders "hascor" shouldn't be coming from idam it would be on a per jurisdiction / case type bases which at the moment would be on our end for the foreseeable future unless CCD would hold some data around this.

as for the judristiction stuff we need to talk with idam to make sure our roles are

casework - {judristiction} - {casetype} - judge and then  attached to the judge who will be using this:
reason we need both {judristiction} - {casetype} is because within definition for their divorce jurisdiction there is divorce and fr case types 

below is the different possible judristiction/casetpye i did in my org pr which would grab the templeates
```
    const templates = {
        sscs: {
            benefit : require('./sscs/benefit')
        },
        cmc: {
            moneyclaimcase: require('./cmc/moneyclaimcase')
        },
        probate: {
            grantofrepresentation: require('./probate/grantofrepresentation')
        },
        divorce: {
            divorce: require('./divorce/divorce'),
            financialremedymvp2: require('./divorce/financialremedy')
        }
    };
``` 
below is also the hardcoing i did to match up per jurisdiction and case type
```
module.exports = {
    sscsbenfits: {
        jud: 'SSCS',
        caseType: 'Benefit',
        coh: true,
        filter: '&state=appealCreated&case.appeal.benefitType.code=PIP'
    },
    cmc: {
        jud: 'CMC',
        caseType: 'MoneyClaimCase',
        coh: false,
        filter: ''
    },
    probate: {
        jud: 'PROBATE',
        caseType: 'GrantOfRepresentation',
        coh: false,
        filter: ''
    },
    divorce: {
        jud: 'DIVORCE',
        caseType: 'DIVORCE',
        coh: false,
        filter: ''
    },
    financialremedy: {
        jud: 'DIVORCE',
        caseType: 'FinancialRemedyMVP2',
        coh: false,
        filter: ''
    }
};
```